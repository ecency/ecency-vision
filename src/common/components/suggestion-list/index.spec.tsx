import React from "react";

import SuggestionList from "./index";

import renderer from "react-test-renderer";

const props = {
  items: ["foo", "bar", "baz"],
  // header: "My List",
  // renderer: (i: any) => <> {`# ${i}`}</>,
  modeItems: [{
    items: ["foo", "bar", "baz"],
    header: "My List",
    renderer: (i: any) => <> {`# ${i}`}</>,
  }]
};

const component = renderer.create(
  <SuggestionList {...props}>
    <span>children</span>
  </SuggestionList>
);

it("(1) Default render", () => {
  expect(component.toJSON()).toMatchSnapshot();
});

it("(2) Show list", () => {
  const instance: any = component.getInstance();
  instance.setState({ showList: true });
  expect(component.toJSON()).toMatchSnapshot();
});
