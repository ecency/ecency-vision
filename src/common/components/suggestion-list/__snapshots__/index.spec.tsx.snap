// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Default render 1`] = `
<div
  className="suggestion relative"
>
  <span>
    children
  </span>
  <div />
</div>
`;

exports[`(2) Show list 1`] = `
<div
  className="suggestion relative"
>
  <span>
    children
  </span>
  <div>
    <div
      className="suggestion-list-parent"
    >
      <div
        className="search-suggestion-list"
      >
        <div
          className="list-header"
        >
          My List
        </div>
        <div
          className="list-body"
        >
          <a
            className="list-item"
            href="#"
            onClick={[Function]}
          >
             
            # foo
          </a>
          <a
            className="list-item"
            href="#"
            onClick={[Function]}
          >
             
            # bar
          </a>
          <a
            className="list-item"
            href="#"
            onClick={[Function]}
          >
             
            # baz
          </a>
        </div>
      </div>
      <div
        className="search-suggestion-list more-result"
      >
        <div
          className="list-body"
        >
          <a
            className="list-item"
            href="#"
            onClick={[Function]}
          >
            More results
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
